src/panels/calendar/ha-panel-calendar.js

```diff
@@ -0,0 +1,180 @@
+            <paper-listbox id="calendar_list" multi on-selected-items-changed="_fetchData" selected-values="{{selectedCalendars}}" attr-for-selected="item-name">
+              <template is="dom-repeat" items="[[calendars]]">
+                <paper-item item-name="[[item.entity_id]]">
+                  <span class="calendar_color" style$="background-color: [[item.color]]"></span>
+                  <span class="calendar_color_spacer"></span>
+                  [[item.name]]
+                </paper-item>
+              </template>
+            </paper-listbox>
+          </paper-card>
+        </div>
+        <div class="flex layout horizontal wrap">
+          <ha-big-calendar date-Updated="[[dateUpdated()]]" events="[[events]]"></ha-big-calendar>
+        </div>
+      </div>
+    </app-header-layout>
+    `;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._fetchData = this._fetchData.bind(this);
+    // TODO implement calendar_updated event
+    // this.hass.connection.subscribeEvents(this._fetchData, 'calendar_updated')
+    //  .then(function (unsub) { this._unsubEvents = unsub; }.bind(this));
+    const now = new Date();
+    this.start = new Date(now.getFullYear(), now.getMonth(), 1).toISOString();
+    this.end = new Date(now.getFullYear(), now.getMonth() + 1, 1).toISOString();
+
+    this._fetchData();
+  }
+
+  _fetchData() {
+    if (this.start == null || this.end == null) {
+      return;
+    }
+    // Fetch calendar list
+    this.hass.callApi('get', 'calendars')
+      .then((result) => {
+        this.calendars = result;
+      });
+    // Fetch events for selected calendar
+    const params = encodeURI(`?start=${this.start}&end=${this.end}`);
+    const calls = this.selectedCalendars.map(cal => this.hass.callApi('get', `calendar/${cal}${params}`));
+    Promise.all(calls).then((results) => {
+      var tmpEvents = [];
+
+      for (let i = 0; i < results.length; i++) {
+        tmpEvents = tmpEvents.concat(results[i]);
+      }
+
+      for (let i = 0; i < tmpEvents.length; i++) {
+        tmpEvents[i].start = new Date(tmpEvents[i].start);
+        if (tmpEvents[i].end) {
+          tmpEvents[i].end = new Date(tmpEvents[i].end);
+        } else {
+          tmpEvents[i].end = null;
+        }
+      }
+      this.events = tmpEvents;
+    });
+  }
+
+  checkAll() {
+    // Check all calendars
+    if (this.$.all_calendars.checked) {
+      const selectedIndex = this.$.calendar_list.selectedItems
+        .map(x => this.$.calendar_list.indexOf(x));
+      for (let i = 0; i < this.calendars.length; i++) {
+        if (selectedIndex.indexOf(i) === -1) {
+          this.$.calendar_list.selectIndex(i);
+        }
+      }
+    }
+  }
+
+  dateUpdated() {
+    var pol = this;
+    return function (startDate) {
+      pol.start = new Date(startDate.getFullYear(), startDate.getMonth(), 1).toISOString();
+      pol.end = new Date(startDate.getFullYear(), startDate.getMonth() + 1, 1).toISOString();
```

Comment:
why is the duration always 1 month?
shouldn't you get this duration from the calendar?

Reply:
We should get the duration from the current view using the `onView` callback from BigCalendar. We can then get the range by calling `currentView.range(startDate)`

Reply:
@balloob I think it's not so easy : https://github.com/intljusticemission/react-big-calendar/issues/342
So I guess I need to add the current view and `current date` and current_view` properties in the ha-panel-calendar component